File: WADV.Core/IScriptEngine.vb
''' <summary>
''' 脚本引擎接口
''' </summary>
Public Interface IScriptEngine

    ''' <summary>
    ''' 初始化脚本引擎
    ''' </summary>
    Sub Initialise()
    ''' <summary>
    ''' 运行一个脚本文件
    ''' </summary>
    ''' <param name="filePath"></param>
    Sub RunFileAsync(filePath As String)
    ''' <summary>
    ''' 运行一个脚本文件
    ''' </summary>
    ''' <param name="filePath"></param>
    ''' <returns></returns>
    Function RunFile(filePath As String) As Object
    ''' <summary>
    ''' 运行一串脚本代码
    ''' </summary>
    ''' <param name="content"></param>
    Sub RunStringAsync(content As String)
    ''' <summary>
    ''' 运行一串脚本代码
    ''' </summary>
    ''' <param name="content"></param>
    ''' <returns></returns>
    Function RunString(content As String) As Object
    ''' <summary>
    ''' 设置脚本全局变量的值
    ''' </summary>
    ''' <param name="name">变量名</param>
    ''' <param name="value">目标值</param>
    Sub [Set](name As String, value As Object)
    ''' <summary>
    ''' 获取脚本全局变量的值
    ''' </summary>
    ''' <param name="name">变量名</param>
    ''' <returns></returns>
    Function [Get](name As String) As Object

    ''' <summary>
    ''' 释放脚本引擎的资源
    ''' </summary>
    Sub Dispose()
End Interface

File: WADV.Core/Script/IScriptEngine.cs
using System;

namespace WADV.Core.Script {
    public interface IScriptEngine : IDisposable {
        /// <summary>
        /// Initialize engine
        /// </summary>
        void Initialize();

        /// <summary>
        /// Run script file in new thread
        /// </summary>
        /// <param name="filePath">Script file location</param>
        void RunFileAsync(string filePath);

        /// <summary>
        /// Run script file
        /// </summary>
        /// <param name="filePath">Script file location</param>
        /// <returns></returns>
        object RunFile(string filePath);

        /// <summary>
        /// Run script code in new thread
        /// </summary>
        /// <param name="content">Code content</param>
        void RunStringAsync(string content);

        /// <summary>
        /// Run script code
        /// </summary>
        /// <param name="content">Code content</param>
        /// <returns></returns>
        object RunString(string content);

        /// <summary>
        /// Set value of a global object
        /// </summary>
        /// <param name="name">Object name</param>
        /// <param name="value">Object value</param>
        void Set(string name, object value);

        /// <summary>
        /// Get value of a global object
        /// </summary>
        /// <param name="name">Object name</param>
        /// <returns></returns>
        object Get(string name);

        /// <summary>
        /// Get value of a global object
        /// </summary>
        /// <typeparam name="T">Object type</typeparam>
        /// <param name="name">Object name</param>
        /// <returns></returns>
        T Get<T>(string name);

        /// <summary>
        /// Register Cross Language Linking Field
        /// </summary>
        /// <param name="target">Target field</param>
        void Register(Field target);

        /// <summary>
        /// Clear all data in this engine
        /// </summary>
        void ClearStatus();
    }
}
